### DISCUSSION TITLE
Conditional modifications of arrays for JSON schemas

### DISCUSSION THREAD

**Discussion by Markus-Elfring** [json-schema-org/community#277] [Q&A]
Some key words are described in the way that they can be mapped to arrays.

Examples:
* enum
* required
* type

The corresponding arrays can contain a varying number of items.
:thought_balloon: I imagine that the item selection will occasionally be constructed by [special case distinctions](https://json-schema.org/understanding-json-schema/reference/combining.html "Schema Composition").

How good are adjustments (like the following) supported by available software tools?
* Adding
* Deleting
* Overwriting

**Comment by handrews** [comment]
JSON Schema does not modify the instance.  It sounds like you want [JSON Patch](https://datatracker.ietf.org/doc/html/rfc6902).

**Reply by Markus-Elfring** [reply]
>It sounds like you want JSON Patch.

Patching of JSON files can also be interesting.

:eyes: But I would like to take another look at possibilities for schema composition.


>JSON Schema does not modify the instance.

:thinking:
* Will such information trigger any further development considerations?
* How would you like to adjust array contents on demand according to design requirements of customised data models?
